scalableminds/webknossos
# Self-Hosted User Datastores

### URL of deployed dev instance (used for testing):
- https://autoupdate.webknossos.xyz

### Steps to test:
- build datastore jar with `sbt "project webknossosDatastore" assembly`
- copy datastore jar into `tools/auto_update` folder and rename it to `webknossos.jar`
- execute `run.sh`
- => datastore should startup, check for updates and restart itself after downloading the update
- => after initial update occurs, no further restarts should happen
- => if `autoUpdate.enabled` is `false`, no update should happen
- to test the install script, send a `POST` request to `/api/userDatastores` with the following body fields `{url: string, port: int, name: string}`
- => should return a custom install script that guides you through the setup process 

### Issues:
- fixes #5579 

------
